src/main/scala/shine/cuda/ast/Printer.scala

```diff
@@ -0,0 +1,80 @@
+package shine.cuda.ast
+
+import arithexpr.arithmetic.ArithExpr
+import shine.C.AST._
+import shine.DPIA.Types.AddressSpace
+import shine.{OpenCL, cuda}
+import shine.OpenCL.AST.KernelDecl
+import shine.cuda.BuiltInAttribute
+
+object Printer {
+  def apply(n: Node): String = (new Printer).printNode(n)
+}
+
+class Printer extends shine.OpenCL.AST.Printer {
+  override def printStmt(s: Stmt): Unit = s match {
+    case _: SynchronizeThreads => print("__syncthreads();")
+    case _: SynchronizeWarp => print("__syncwarp();")
+    case _ => super.printStmt(s)
+  }
+
+  override def toString(e: ArithExpr): String = e match {
+    case of: BuiltInAttribute => of.toString
+
+    case _ => super.toString(e)
+  }
+
+  override def printKernelDecl(k: KernelDecl): Unit = {
+    //TODO
+    println("#include <mma.h>")
+
+    print("extern \"C\" __global__")
+    println("")
+    print(s"void ${k.name}(")
+    k.params.foreach(p => {
+      printDecl(p)
+      if (!p.eq(k.params.last)) print(", ")
+    })
+    print(")")
+
+    printStmt(k.body)
+  }
+
+  override def printDecl(d: Decl): Unit = {
+    d match {
+      case v: shine.OpenCL.AST.VarDecl if v.t.isInstanceOf[ExternArrayType] =>
+        print("extern ")
+      case _ =>
+    }
+
+    super.printDecl(d)
+  }
+
+
+  override def printParamDecl(p: ParamDecl): Unit = {
+    if (p.t.const) print("const ")
+    p.t match {
+      case b: BasicType => print(s"${b.print} ${p.name}")
```

Comment:
why is it `b.print` now?

Reply:
same reason as the change in the C/AST/Printer.
But i think the change here to `b.print` should be sufficient (so the change in C/AST/Printer is not necessary)